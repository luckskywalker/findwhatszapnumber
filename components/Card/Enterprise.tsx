import {IEnterprise} from "@/typescript/enterprise";
import Image from "next/image";
import React from "react";

type TEnterpriseProps = {
    className?: string,
    enterprise: IEnterprise
}

const Enterprise: React.FC<TEnterpriseProps> = ({className, enterprise}) => {
  return (
    <div className={`flex flex-col sm:w-96 w-full bg-white border shadow-sm rounded-xl dark:bg-neutral-900 dark:border-neutral-700 dark:shadow-neutral-700/70 ${className}`}>
      <Image width={270} height={150} className="rounded-t-xl sm:w-96 w-full" src={`data:image/${enterprise?.image?.contentType};base64,${enterprise?.image?.data?.toString("base64")}`} alt="Card Image" />
      <div className="p-4 md:p-5">
        <h3 className="text-lg font-bold text-gray-800 dark:text-white">
          {enterprise.label}
        </h3>
        <p className="mt-1 text-gray-500 dark:text-neutral-400 break-words">
          {enterprise.description}
        </p>
        <a className="mt-2 py-2 px-3 inline-flex justify-center items-center gap-x-2 text-sm font-medium rounded-lg border border-transparent bg-blue-600 text-white hover:bg-blue-700 focus:outline-none focus:bg-blue-700 disabled:opacity-50 disabled:pointer-events-none" href="#">
          {enterprise.phoneNumber}
        </a>
      </div>
    </div>
  );
};

export default Enterprise;
