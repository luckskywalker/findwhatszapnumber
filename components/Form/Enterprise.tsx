"use client";

import React from "react";
import {useForm, SubmitHandler, DeepPartial, useController} from "react-hook-form";
import SelectAddress from "@/components/Select/Address";
import Field from "@/components/Field";

import {yupResolver} from "@hookform/resolvers/yup";
import * as yup from "yup";
import {InferType} from "yup";

const enterpriseSchema = yup.object({
  label: yup.string().required(),
  description: yup.string().required(),
  phoneNumber: yup.string().required(),
  address: yup.string().required(),
  image: yup.mixed(),
  category: yup.string().required()
});

type TEnterprise = InferType<typeof enterpriseSchema> 

type TFormEnterpriseProps = {
    defaultValues?: DeepPartial<TEnterprise>,
    onSubmit: SubmitHandler<TEnterprise>
}

type TFormEnterpriseConnectedProps = {
    defaultValues?: DeepPartial<TEnterprise>,
    onError?: (e: unknown) => void,
    onSuccess?: (data: unknown) => void

}

const FormEnterprise: React.FC<TFormEnterpriseProps> = ({defaultValues, onSubmit, ...props}) => {
  const {register, handleSubmit, control, formState: {errors}} = useForm<TEnterprise>({resolver: yupResolver(enterpriseSchema), defaultValues});
  const {field} = useController({control, name: "address"});

  return (
    <form onSubmit={handleSubmit(onSubmit)} {...props}>
      <Field error={errors.label}>
        <input className="input input-bordered w-full"
          type="text"
          placeholder="label"
          aria-label="label"
          {...register("label")}
        />
      </Field>
      <Field error={errors.description}>
        <textarea className="textarea textarea-bordered w-full" rows={3} placeholder="description" aria-label="description" {...register("description")} />
      </Field>
      <Field error={errors.category}>
        <select className="select select-bordered w-full" aria-label="category" {...register("category")} >
          <option value="Coiffeur">Coiffeur</option>
          <option value="Maçon">Maçon</option>
          <option value="Restaurant">Restaurant</option>
        </select>
      </Field>
      <Field error={errors.phoneNumber}>
        <input className="input input-bordered w-full" type="text" placeholder="phone number" aria-label="phoneNumber" {...register("phoneNumber")} />
      </Field>
      <Field error={errors.address}>
        <SelectAddress aria-label="address" {...field} />
      </Field>
      <Field error={errors.image}>
        <input type="file" className="file-input input-bordered w-full" placeholder="image" aria-label="image" {...register("image")} />
      </Field>
      <button
        type="submit"
        className="btn btn-primary w-full"
        aria-label="submit"
      >
        Update Enterprise
      </button>
      
    </form >
  );
};

const FormEnterpriseConnected: React.FC<TFormEnterpriseConnectedProps> = ({defaultValues, onError, onSuccess}) => {

  const handleSubmit = async (data: TEnterprise) => {
    let resjson: TEnterprise | null = null;

    const tosenddata = new FormData();

    for (const key in data) {
      tosenddata.append(key, data[key as keyof TEnterprise] as string);
    }

    if (data.image && data.image[0]) {
      tosenddata.set("image", data.image[0]);
    }

    try {
      const res = await fetch("/api/user/enterprise/update", {
        method: "POST",
        body: tosenddata
      });
      resjson = await res.json();
    } catch (e) {
      onError && onError(e);
    } finally {
      onSuccess && onSuccess(resjson);
    }
  };


  return <FormEnterprise onSubmit={handleSubmit} defaultValues={defaultValues}/>;
};


export default FormEnterpriseConnected;