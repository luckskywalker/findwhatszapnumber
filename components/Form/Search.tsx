import SelectCategory from "@/components/Select/Category";
import SelectAddress from "@/components/Select/Address";

type TSearchFormProps = {
}

const SearchForm: React.FC<TSearchFormProps> = ({}) => {

  return (
    <form action="/">
      <div className="mb-4">
        <input className="input input-bordered w-full"
          type="text"
          name="keywords"
          placeholder="keywords"
        />
      </div>
      <div className="mb-4">
        <SelectAddress name="address" className="w-full"/>
      </div>
      <div className="mb-4">
        <SelectCategory className="w-full"/>
      </div>
      <div className="mb-4">
        <button
          type="submit"
          className="btn btn-primary w-full"
        >
          Search
        </button>
      </div>
    </form >
  );
};




export default SearchForm;