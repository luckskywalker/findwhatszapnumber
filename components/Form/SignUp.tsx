"use client";

import {useForm, SubmitHandler} from "react-hook-form";
import Field from "@/components/Field";
import {yupResolver} from "@hookform/resolvers/yup";
import * as yup from "yup";
import {InferType} from "yup";
import {signIn} from "next-auth/react";


const userSchema = yup.object({
  email: yup.string().email().required(),
  password: yup.string().min(8).required(),
  passwordCheck: yup.string().required().oneOf([yup.ref("password")], "Passwords must match"),
});

type TUserSignup = InferType<typeof userSchema>

type TSignUpProps = {
  onSubmit: SubmitHandler<TUserSignup>
}

type TSignUpConnectedProps = {
  onError?: (e: unknown) => void, 
  onSuccess?: (data: TUserSignup) => void
}


const FormSignUp: React.FC<TSignUpProps> = ({onSubmit}) => {
  const {register, handleSubmit, formState: {errors}, watch} = useForm<TUserSignup>({resolver: yupResolver(userSchema)});

  return (
    <form onSubmit={handleSubmit(onSubmit)}>
      
      <Field error={errors.email}>
        <input className="input input-bordered w-full"
          type="email"
          placeholder="email"
          {...register("email")}
        />
      </Field>
      
      
      <Field error={errors.password}>
        <input className="input input-bordered w-full"
          type="password"
          placeholder="password"
          {...register("password")}
        />
      </Field>
      
      <Field error={errors.passwordCheck}>
        <input className="input input-bordered w-full"
          type="password"
          placeholder="cornfirm password"
          {...register("passwordCheck")}
        />
      </Field>
      <button
        type="submit"
        className="btn btn-primary w-full"
      >
        Sign Up
      </button>
      
    </form >
  );
};


const FormSignUpConnected: React.FC<TSignUpConnectedProps> = ({onError= (data) => console.log(data), onSuccess = (data) => console.log(data)}) => {

  const handleSubmit = async (data: TUserSignup) => {
    let resjson: TUserSignup | null = null;

    const tosenddata = new FormData();

    for (const key in data) {
      tosenddata.append(key, data[key as keyof TUserSignup] as string);
    }
    await signIn("Credentials", data);
    // try {
    //   const res = await fetch("/api/auth/signup", {
    //     method: "POST",
    //     body: JSON.stringify(data)
    //   });
    //   resjson = await res.json();
    // } catch (e) {
    //   onError && onError(e);
    // } finally {
    //   onSuccess && onSuccess(resjson as TUserSignup);
    // }
  };


  return <FormSignUp onSubmit={handleSubmit} />;
};




export default FormSignUpConnected;