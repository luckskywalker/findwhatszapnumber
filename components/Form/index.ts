import Enterprise from "./Enterprise";
import SignUp from "./SignUp";
import Search from "./Search";

const Form = {
  Enterprise,
  SignUp,
  Search
};

export default Form;
