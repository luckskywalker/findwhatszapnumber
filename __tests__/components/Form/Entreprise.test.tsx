import {render, screen, fireEvent} from "@testing-library/react";
import {FormEnterprise} from "@/components/Form/Enterprise";

describe("Page", () => {

  it("should be ok", async () => {
    render(<FormEnterprise onSubmit={() => console.log("test")} />);
    await fireEvent.click(screen.getByText(/submit/i));
  });
});