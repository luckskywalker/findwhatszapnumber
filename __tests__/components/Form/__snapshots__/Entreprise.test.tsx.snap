// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Form Enterprise should show errors fields empty 1`] = `
<div>
  <form>
    <div
      class="mb-4"
    >
      <label />
      <input
        aria-label="label"
        class="input input-bordered w-full"
        name="label"
        placeholder="label"
        type="text"
      />
      <p
        class="text-red-500 text-xs italic mt-2"
      >
        label is a required field
      </p>
    </div>
    <div
      class="mb-4"
    >
      <label />
      <textarea
        aria-label="description"
        class="textarea textarea-bordered w-full"
        name="description"
        placeholder="description"
        rows="3"
      />
      <p
        class="text-red-500 text-xs italic mt-2"
      >
        description is a required field
      </p>
    </div>
    <div
      class="mb-4"
    >
      <label />
      <select
        aria-label="category"
        class="select select-bordered w-full"
        name="category"
      >
        <option
          value="Coiffeur"
        >
          Coiffeur
        </option>
        <option
          value="Maçon"
        >
          Maçon
        </option>
        <option
          value="Restaurant"
        >
          Restaurant
        </option>
      </select>
    </div>
    <div
      class="mb-4"
    >
      <label />
      <input
        aria-label="phoneNumber"
        class="input input-bordered w-full"
        name="phoneNumber"
        placeholder="phone number"
        type="text"
      />
      <p
        class="text-red-500 text-xs italic mt-2"
      >
        phoneNumber is a required field
      </p>
    </div>
    <div
      class="mb-4"
    >
      <label />
      <input
        hidden=""
        readonly=""
        style="position: fixed; top: 1px; left: 1px; width: 1px; height: 0px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; border-width: 0px; display: none;"
        type="hidden"
      />
      <div
        aria-label="address"
        data-headlessui-state=""
      >
        <input
          aria-autocomplete="list"
          aria-expanded="false"
          aria-label="Address"
          class="input input-bordered w-full"
          data-headlessui-state=""
          id="headlessui-combobox-input-:r2:"
          placeholder="Address"
          role="combobox"
          type="text"
          value=""
        />
      </div>
      <p
        class="text-red-500 text-xs italic mt-2"
      >
        address is a required field
      </p>
    </div>
    <div
      class="mb-4"
    >
      <label />
      <input
        aria-label="image"
        class="file-input input-bordered w-full"
        name="image"
        placeholder="image"
        type="file"
      />
    </div>
    <button
      aria-label="submit"
      class="btn btn-primary w-full"
      type="submit"
    >
      Update Enterprise
    </button>
  </form>
</div>
`;

exports[`Component Form Enterprise should show errors on address only 1`] = `
<div>
  <form>
    <div
      class="mb-4"
    >
      <label />
      <input
        aria-label="label"
        class="input input-bordered w-full"
        name="label"
        placeholder="label"
        type="text"
      />
    </div>
    <div
      class="mb-4"
    >
      <label />
      <textarea
        aria-label="description"
        class="textarea textarea-bordered w-full"
        name="description"
        placeholder="description"
        rows="3"
      />
    </div>
    <div
      class="mb-4"
    >
      <label />
      <select
        aria-label="category"
        class="select select-bordered w-full"
        name="category"
      >
        <option
          value="Coiffeur"
        >
          Coiffeur
        </option>
        <option
          value="Maçon"
        >
          Maçon
        </option>
        <option
          value="Restaurant"
        >
          Restaurant
        </option>
      </select>
    </div>
    <div
      class="mb-4"
    >
      <label />
      <input
        aria-label="phoneNumber"
        class="input input-bordered w-full"
        name="phoneNumber"
        placeholder="phone number"
        type="text"
      />
    </div>
    <div
      class="mb-4"
    >
      <label />
      <input
        hidden=""
        readonly=""
        style="position: fixed; top: 1px; left: 1px; width: 1px; height: 0px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; border-width: 0px; display: none;"
        type="hidden"
      />
      <div
        aria-label="address"
        data-headlessui-state=""
      >
        <input
          aria-autocomplete="list"
          aria-expanded="false"
          aria-label="Address"
          class="input input-bordered w-full"
          data-headlessui-state=""
          id="headlessui-combobox-input-:r8:"
          placeholder="Address"
          role="combobox"
          type="text"
          value=""
        />
      </div>
      <p
        class="text-red-500 text-xs italic mt-2"
      >
        address is a required field
      </p>
    </div>
    <div
      class="mb-4"
    >
      <label />
      <input
        aria-label="image"
        class="file-input input-bordered w-full"
        name="image"
        placeholder="image"
        type="file"
      />
    </div>
    <button
      aria-label="submit"
      class="btn btn-primary w-full"
      type="submit"
    >
      Update Enterprise
    </button>
  </form>
</div>
`;
